File: _designs_001/Implement Queue using Stacks/Solution.java
public class Solution {
    public List<Integer> testMyQueue(StreamTokenizer tokenizer, List<Double> metrics) throws IOException {
        MyQueue myQueue = new MyQueue();
        List<Integer> result = new ArrayList<Integer>();

        int opsCnt = Serializer.deserializeInt(tokenizer);
        for (int i = 0; i < opsCnt; i++) {
            int n;
            double startTime;
            switch (Serializer.deserializeInt(tokenizer)) {
            case 0:
                n = Serializer.deserializeInt(tokenizer);
                startTime = System.nanoTime();
                myQueue.push(n);
                metrics.set(0, metrics.get(0) + (double) (System.nanoTime() - startTime) / 1000000L);
                break;
            case 1:
                startTime = System.nanoTime();
                myQueue.pop();
                metrics.set(0, metrics.get(0) + (double) (System.nanoTime() - startTime) / 1000000L);
                break;
            case 2:
                startTime = System.nanoTime();
                result.add(myQueue.peek());
                metrics.set(0, metrics.get(0) + (double) (System.nanoTime() - startTime) / 1000000L);
                break;
            case 3:
                startTime = System.nanoTime();
                result.add(myQueue.empty() ? 1 : 0);
                metrics.set(0, metrics.get(0) + (double) (System.nanoTime() - startTime) / 1000000L);
                break;
            }
        }
        return result;
    }
}

File: _designs_001/Implement Queue using Stacks/Solution.rb
class Solution
  def testMyQueue(inp, metrics)
    metrics[0] = 0
    q, r = MyQueue.new, []
    Serializer.deserializeInt(inp).times do
      case Serializer.deserializeInt(inp)
      when 0
        n = Serializer.deserializeInt(inp)
        start_time = Time.now
        q.push(n)
        metrics[0] += (Time.now - start_time) * 1000
      when 1
        start_time = Time.now
        q.pop
        metrics[0] += (Time.now - start_time) * 1000
      when 2
        start_time = Time.now
        r << q.peek
        metrics[0] += (Time.now - start_time) * 1000
      when 3
        start_time = Time.now
        r << (q.empty == true ? 1 : 0)
        metrics[0] += (Time.now - start_time) * 1000
      end
    end
    r
  end
end
